src/commands/mod.rs

```diff
@@ -990,39 +990,19 @@ impl DaemonControl {
         limit: u64,
     ) -> ListTransactionsResult {
         let mut db_conn = self.db.connection();
+        // Note the result could in principle be retrieved in a single database query.
         let txids = db_conn.list_txids(start, end, limit);
-        let transactions = txids
-            .iter()
-            .filter_map(|txid| {
-                // TODO: batch those calls to the Bitcoin backend
-                // so it can in turn optimize its queries.
-                self.bitcoin
-                    .wallet_transaction(txid)
-                    .map(|(tx, block)| TransactionInfo {
-                        tx,
-                        height: block.map(|b| b.height),
-                        time: block.map(|b| b.time),
-                    })
-            })
-            .collect();
-        ListTransactionsResult { transactions }
+        self.list_transactions(&txids)
     }
 
     /// list_transactions retrieves the transactions with the given txids.
     pub fn list_transactions(&self, txids: &[bitcoin::Txid]) -> ListTransactionsResult {
```

Comment:
Perhaps for any txs not found in the DB, I could use the Bitcoin backend as a fallback.